pom.xml

```diff
@@ -33,154 +33,397 @@
-    </dependency>
-
-    <dependency>
-      <artifactId>tez-common</artifactId>
-      <groupId>org.apache.tez</groupId>
-      <version>${tez.version}</version>
-    </dependency>
-
-    <dependency>
-      <artifactId>tez-mapreduce</artifactId>
-      <groupId>org.apache.tez</groupId>
-      <version>${tez.version}</version>
-    </dependency>
-
-    <dependency>
-      <groupId>org.apache.hive</groupId>
-      <artifactId>hive-contrib</artifactId>
-      <version>${hive.version}</version>
-      <scope>test</scope>
-    </dependency>
-
-    <dependency>
-      <groupId>org.reflections</groupId>
-      <artifactId>reflections</artifactId>
-      <version>0.9.8</version>
-    </dependency>
-
-    <dependency>
-      <groupId>org.apache.hadoop</groupId>
-      <artifactId>hadoop-mapreduce-client-common</artifactId>
-      <version>${hadoop.version}</version>
-    </dependency>
-
-    <dependency>
-      <groupId>org.apache.hadoop</groupId>
-      <artifactId>hadoop-mapreduce-client-core</artifactId>
-      <version>${hadoop.version}</version>
-    </dependency>
-
-    <dependency>
-      <groupId>com.github.stefanbirkner</groupId>
-      <artifactId>system-rules</artifactId>
-      <version>1.19.0</version>
-      <scope>test</scope>
-    </dependency>
-
-    <!-- Always put this before JUnit or the class loader might load the
-      wrong Matcher -->
-    <dependency>
-      <groupId>org.hamcrest</groupId>
-      <artifactId>hamcrest-all</artifactId>
-      <version>1.3</version>
-      <scope>test</scope>
-    </dependency>
-
-    <dependency>
-      <groupId>org.mockito</groupId>
-      <artifactId>mockito-core</artifactId>
-      <version>${mockito.version}</version>
-      <scope>test</scope>
-    </dependency>
-
-    <dependency>
-      <groupId>org.mockito</groupId>
-      <artifactId>mockito-junit-jupiter</artifactId>
-      <version>${mockito.version}</version>
-      <scope>test</scope>
-    </dependency>
-
-    <dependency>
-      <groupId>junit</groupId>
-      <artifactId>junit</artifactId>
-      <version>4.13.2</version>
-      <scope>provided</scope>
-    </dependency>
-
-    <dependency>
-      <groupId>org.junit.jupiter</groupId>
-      <artifactId>junit-jupiter</artifactId>
-      <version>${junit.jupiter.version}</version>
-    </dependency>
-
-    <dependency>
-      <groupId>org.junit.vintage</groupId>
-      <artifactId>junit-vintage-engine</artifactId>
-      <version>${junit.jupiter.version}</version>
-    </dependency>
-
-  </dependencies>
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <hadoop.version>3.4.0</hadoop.version>
+        <hive.version>4.0.1</hive.version>
+        <hive.execution.engine>tez</hive.execution.engine>
+        <junit.jupiter.version>5.11.2</junit.jupiter.version>
+        <junit.version>4.13.2</junit.version>
+        <tez.version>0.10.4</tez.version>
+        <mockito.version>5.14.2</mockito.version>
```

Comment:
Can you put all the versions that aren't used more than once back down into the `dependency` declaration? Up to now the only ones declared as properties are the ones which are repeated.

Reply:
I can do that bu it the same way this was added before my changes:

<license.maven.plugin.version>3.0</license.maven.plugin.version>

so I do not see a one approach here/

Reply:
Done.